clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/AbstractAvroStoreClient.java

```diff
@@ -133,8 +133,11 @@ protected AbstractAvroStoreClient(
     this.compressorFactory = new CompressorFactory();
     this.storageRequestPath = TYPE_STORAGE + "/" + clientConfig.getStoreName();
     this.computeRequestPath = TYPE_COMPUTE + "/" + clientConfig.getStoreName();
-    this.veniceClientWarmUpExecutor = Executors
-        .newFixedThreadPool(1, new DaemonThreadFactory("Venice-Client-Warmup-For-" + clientConfig.getStoreName()));
+    this.veniceClientWarmUpExecutor = Optional.ofNullable(clientConfig.getWarmupExecutor())
```

Comment:
Could you helpt o add a unit test to verify this work as expected? Example: `AbstractAvroStoreClientTest`

Reply:
Thanks - added a test case to verify the executor is used when provided.